Source/WebKit/Shared/WebCompiledContentRuleListData.cpp

```diff
@@ -32,97 +32,31 @@
-    encoder << topURLFiltersBytecodeOffset;
-    encoder << topURLFiltersBytecodeSize;
-    encoder << frameURLFiltersBytecodeOffset;
-    encoder << frameURLFiltersBytecodeSize;
+    return data->createHandle(protection);
 }
 
-std::optional<WebCompiledContentRuleListData> WebCompiledContentRuleListData::decode(IPC::Decoder& decoder)
+WebCompiledContentRuleListData::WebCompiledContentRuleListData(String&& identifier, std::optional<WebKit::SharedMemoryHandle>&& dataHandle, size_t actionsOffset, size_t actionsSize, size_t urlFiltersBytecodeOffset, size_t urlFiltersBytecodeSize, size_t topURLFiltersBytecodeOffset, size_t topURLFiltersBytecodeSize, size_t frameURLFiltersBytecodeOffset, size_t frameURLFiltersBytecodeSize)
+    : identifier(WTFMove(identifier))
+    , data(dataHandle ? SharedMemory::map(WTFMove(*dataHandle), SharedMemory::Protection::ReadOnly) : nullptr)
+    , actionsOffset(actionsOffset)
+    , actionsSize(actionsSize)
+    , urlFiltersBytecodeOffset(urlFiltersBytecodeOffset)
+    , urlFiltersBytecodeSize(urlFiltersBytecodeSize)
+    , topURLFiltersBytecodeOffset(topURLFiltersBytecodeOffset)
+    , topURLFiltersBytecodeSize(topURLFiltersBytecodeSize)
+    , frameURLFiltersBytecodeOffset(frameURLFiltersBytecodeOffset)
+    , frameURLFiltersBytecodeSize(frameURLFiltersBytecodeSize)
 {
-    std::optional<String> identifier;
-    decoder >> identifier;
-    if (!identifier)
-        return std::nullopt;
-
-    std::optional<std::optional<SharedMemory::Handle>> handle;
-    decoder >> handle;
-    if (!handle)
-        return std::nullopt;
-
-    std::optional<size_t> actionsOffset;
-    decoder >> actionsOffset;
-    if (!actionsOffset)
-        return std::nullopt;
-
-    std::optional<size_t> actionsSize;
-    decoder >> actionsSize;
-    if (!actionsSize)
-        return std::nullopt;
-
-    std::optional<size_t> urlFiltersBytecodeOffset;
-    decoder >> urlFiltersBytecodeOffset;
-    if (!urlFiltersBytecodeOffset)
-        return std::nullopt;
-
-    std::optional<size_t> urlFiltersBytecodeSize;
-    decoder >> urlFiltersBytecodeSize;
-    if (!urlFiltersBytecodeSize)
-        return std::nullopt;
-
-    std::optional<size_t> topURLFiltersBytecodeOffset;
-    decoder >> topURLFiltersBytecodeOffset;
-    if (!topURLFiltersBytecodeOffset)
-        return std::nullopt;
-
-    std::optional<size_t> topURLFiltersBytecodeSize;
-    decoder >> topURLFiltersBytecodeSize;
-    if (!topURLFiltersBytecodeSize)
-        return std::nullopt;
-
-    std::optional<size_t> frameURLFiltersBytecodeOffset;
-    decoder >> frameURLFiltersBytecodeOffset;
-    if (!frameURLFiltersBytecodeOffset)
-        return std::nullopt;
-
-    std::optional<size_t> frameURLFiltersBytecodeSize;
-    decoder >> frameURLFiltersBytecodeSize;
-    if (!frameURLFiltersBytecodeSize)
-        return std::nullopt;
-    
-    if (!*handle)
-        return std::nullopt;
-    auto data = SharedMemory::map(WTFMove(**handle), SharedMemory::Protection::ReadOnly);
-    if (!data)
```

Comment:
Crash logs actually indicate this was a cause of extremely rare "decode failure" crashes when the mapping fails.  This will probably fix those crashes too.